src/main/java/seedu/address/model/student/IsFreePredicate.java

```diff
@@ -0,0 +1,69 @@
+package seedu.address.model.student;
+
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.model.module.Day;
+import seedu.address.model.module.Timing;
+
+/**
+ * Tests that a {@code Student} is free between a given timing
+ */
+public class IsFreePredicate implements Predicate<Student> {
+    private final Timing startTime;
+    private final Timing endTime;
+
+    private final Day day;
+
+    /**
+     * Constructor to create a predicate
+     * @param startTime starting time of range
+     * @param endTime ending time of range
+     * @param day day to check
+     */
+    public IsFreePredicate(Timing startTime, Timing endTime, Day day) {
+        this.startTime = startTime;
+        this.endTime = endTime;
+        this.day = day;
+    }
+
+    @Override
+    public boolean test(Student student) {
+        return student.getModuleTimings()
+                .stream()
+                .filter(
+                        moduleTiming -> moduleTiming.getDay().equals(day)
+                )
+                .noneMatch(
+                        moduleTiming ->
+                                startTime.compareTo(moduleTiming.getEndTime()) == -1
+                                && endTime.compareTo(moduleTiming.getStartTime()) == 1
+                );
```

Comment:
Very cool use of streams! The compareTo part maybe should be < 0 instead of == -1 (viceversa for the ==1 part), according to Sam Patrick Floyd from StackOverflow

![image](https://github.com/AY2324S2-CS2103T-W12-2/tp/assets/44725033/8f462e7f-2507-4086-afc0-4eb43991d5b9)